### DISCUSSION TITLE
Cable Bulk Import

### DISCUSSION THREAD

**Discussion by Hawk-3** [netbox-community/netbox#7571] [Q&A]
Hello! I'm trying to import (.../dcim/cables/import/):

`side_a_device,side_a_type,side_a_name,side_b_device,side_b_type,side_b_name
PP-A01-01,circuits.circuittermination,LINK-101,PP-B01-01,circuits.circuittermination,LINK-102`

And I get the error:

`<class 'django.core.exceptions.FieldError'>
Cannot resolve keyword 'device' into field. Choices are: _cable_peer, _cable_peer_id, _cable_peer_type, _cable_peer_type_id, _cabled_as_a, _cabled_as_b, cable, cable_id, circuit, circuit_id, created, description, id, last_updated, mark_connected, object_changes, port_speed, pp_info, provider_network, provider_network_id, site, site_id, term_side, upstream_speed, xconnect_id`

`Python version: 3.8.10
NetBox version: 3.0.7`

What could be the problem?

**Comment by candlerb** [comment]
Do you really want to connect a circuit termination back-to-back to another circuit termination?  Given the names "PP-A01-01" (which sound like patch panels), do you mean `dcim.frontport` or `dcim.rearport` instead?

If you can provide the backtrace, it may help to understand what's going on (set EMAIL and ADMINS in configuration.py and it will be E-mailed to you).  It is certainly a bug that an exception is raised instead of a proper error message, but I can't see how to reproduce it.  Can you provide the exact steps?

I tried like this:

* Create circuit XXX, with termination A at my main site
* Create circuit YYY, with termination A at my main site
* Go to connections > cables, Upload

Unfortunately I don't know how to specify circuit terminations in a CSV upload, because circuit terminations don't have names.  I tried:

```
side_a_device,side_a_type,side_a_name,side_b_device,side_b_type,side_b_name
XXX,circuits.circuittermination,A,YYY,circuits.circuittermination,A
```

This just said "Row 1 side_a_device: Object not found." and "Row 1 side_b_device: Object not found."  Same if I replace "A" with the site name. (This appears to be the error associated with `'invalid_choice'`)  

In the [source code](https://github.com/netbox-community/netbox/blob/v3.0.7/netbox/dcim/forms/bulk_import.py#L843-L849) I can't see how CSV upload can identify a cable endpoint unless it has "device" and "name" attributes.

I can manually create a cable between the two circuit terminations though.
